pkg/signature/payload/payload.go

```diff
@@ -74,7 +75,11 @@ func (p Cosign) SimpleContainerImage() SimpleContainerImage {
 
 // MarshalJSON marshals the container signature into a []byte of JSON data
 func (p Cosign) MarshalJSON() ([]byte, error) {
-	return json.Marshal(p.SimpleContainerImage())
+	image := p.SimpleContainerImage()
+	if p.Identity != "" {
+		image.Critical.Identity.DockerReference = p.Identity
```

Comment:
Why is this not located in `SimpleContainerImage`, and specific to `MarshalJSON`?

Reply:
Moved the whole logic into `SimpleContainerImage` as suggested. :+1: